src/awkward/operations/convert.py

```diff
@@ -3403,8 +3692,8 @@ def _parquet_partition_values(path):
 def _parquet_partitions_to_awkward(paths_and_counts):
     path, count = paths_and_counts[0]
     columns = [column for column, value in _parquet_partition_values(path)]
-    values = [[] for column in columns]
-    indexes = [[] for column in columns]
+    values = [[]] * len(columns)
+    indexes = [[]] * len(columns)
```

Comment:
This raises loud warning bells: if you say `[[]] * len(columns)`, then you get a list of _references to the same empty list_, but if you say `[[] for column in columns]`, then you get a list of _distinct empty lists_. That matters if those empty lists are going to be filled, and it looks like that happens below.

If the original code relied upon the lists being independent—appending to one does not append the same item to all lists—which is very likely, then it would fail if it were tested for more than one column. It's quite possible, though, that the tests don't involve more than one column and the fact that this worked shows that we're missing such a test.

```python
>>> not_what_you_want = [[]] * 5
>>> not_what_you_want
[[], [], [], [], []]
>>> not_what_you_want[0].append("wee!")
>>> not_what_you_want
[['wee!'], ['wee!'], ['wee!'], ['wee!'], ['wee!']]
```

Reply:
(As usual, mutation in-place is the source of most subtleties.)

Reply:
Yes quite; I never use it in my own code for the same reasons (I also don't like it for the fact that `[]*n` isn't hugely intuitive even if it is a long term well-known feature. A senior moment I think!

Reply:
If this is a part of "I've fixed the issues with the tests" (your most recent force-push), then you can resolve this now.

I'm also glad that we have tests to catch it: this would have been a bad one.